internal/fs/fs.go

```diff
@@ -2090,20 +2090,27 @@ func (fs *fileSystem) renameHierarchicalDir(ctx context.Context, oldParent inode
 		return err
 	}
 
+	oldDirName := inode.NewDirName(oldParent.Name(), oldName)
+	newDirName := inode.NewDirName(newParent.Name(), newName)
+
 	// If the call for getBucketDirInode fails it means directory does not exist.
 	newDirInode, err := fs.getBucketDirInode(ctx, newParent, newName)
 	if err == nil {
 		// If the directory exists, then check if it is empty or not.
 		if err = fs.checkDirNotEmpty(newDirInode, newName); err != nil {
 			return err
 		}
+
+		// This refers to an empty destination directory.
+		// The RenameFolder API does not allow renaming to an existing empty directory.
+		// As a workaround, we delete the directory first and then perform rename.
```

Comment:
As a workaround, we delete the directory first and then perform rename. >>> To make this work, we delete the empty directory first from gcsfuse and then perform rename.

Reply:
Done